jvm/core/src/main/scala/org/scalatest/events/Event.scala

```diff
@@ -281,6 +281,23 @@ sealed abstract class Event extends Ordered[Event] with Product with Serializabl
       }
     }
   }
+
+  private[events] def serializeRoundtrip(a: Any): Boolean = {
+    try {
+      val baos = new java.io.ByteArrayOutputStream
+      val oos = new java.io.ObjectOutputStream(baos)
+      oos.writeObject(a)
+      oos.flush()
+      val ois = new java.io.ObjectInputStream(new java.io.ByteArrayInputStream(baos.toByteArray))
+      ois.readObject
+      true
+    }
+    catch {
+      case _: Throwable => false
+    }
```

Comment:
@cheeseng We shouldn't catch Throwable, but something more specific. This will turn an OutOfMemoryError into false, for example. Any reason this isn't catching NotSerializableException?